web/src/engine/websites/ProjectSuki.ts

```diff
@@ -0,0 +1,72 @@
+import { Tags } from '../Tags';
+import icon from './ProjectSuki.webp';
+import { DecoratableMangaScraper, Manga, type MangaPlugin } from '../providers/MangaPlugin';
+import * as Common from './decorators/Common';
+import { FetchWindowScript } from '../platform/FetchProvider';
+
+const pageScript = `
+    new Promise ((resolve, reject) => {
+        try {
+
+            const element = document.querySelector('.strip-reader');
+            const bookid = window.location.href.split('/') [4];
```

Comment:
Consider use of spreading `const [, , , , bookid, chapterid ] = window.location.href.split('/');`